airflow/models/taskinstance.py

```diff
@@ -927,6 +929,10 @@ def signal_handler(signum, frame):
                 start_time = time.time()
 
                 self.render_templates(context=context)
+                if STORE_SERIALIZED_DAGS:
+                    RTIF.write(RTIF(ti=self, render_templates=False))
+                    RTIF.delete_old_records(self.task_id, self.dag_id)
```

Comment:
```suggestion
                    RTIF.delete_old_records(self.task_id, self.dag_id, session=session)
```

Reply:
I was purposely not re-using the session for **delete** as the object may still remain in the session until  `commit()` is called, or until we explicitly use expire_all().

WDYT?

Reply:
Is it a problem it hanging around until after the session is commited?

Reply:
Fair point !

Reply:
Updated